### PR TITLE
CHIA-736 Annotate trade_manager.py

### PR DESCRIPTION
**Pull request by AmineKhaldi**

### CODE REVIEW

File: chia/wallet/trade_manager.py [hunk 1]
```diff
@@ -337,6 +340,7 @@ async def cancel_pending_offers(
                         all_txs.append(dataclasses.replace(tx, spend_bundle=None))
                 else:
                     # ATTENTION: new_wallets
+                    assert isinstance(wallet, (CATWallet, DataLayerWallet, NFTWallet))
```
Comment (Quexington) [inline chia/wallet/trade_manager.py:343]: I'm curious is `not isinstance(wallet, Wallet)` would do the trick here?

File: chia/wallet/trade_manager.py [hunk 2]
```diff
@@ -484,10 +488,12 @@ async def _create_offer_for_ids(
                     memos: List[bytes] = []
                     if isinstance(id, int):
                         wallet_id = uint32(id)
-                        wallet = self.wallet_state_manager.wallets[wallet_id]
+                        wallet = self.wallet_state_manager.wallets.get(wallet_id)
+                        assert isinstance(wallet, (CATWallet, Wallet))
                         p2_ph: bytes32 = await wallet.get_puzzle_hash(new=not tx_config.reuse_puzhash)
                         if wallet.type() != WalletType.STANDARD_WALLET:
                             if callable(getattr(wallet, "get_asset_id", None)):  # ATTENTION: new wallets
+                                assert isinstance(wallet, CATWallet)
```
Comment (Quexington) [inline chia/wallet/trade_manager.py:496]: This should be present on all wallets that are not `Wallet` type too (or at least that was the intention).  If this is not the case, then this is fine, but if it is the case and the `not isinstance` trick from above works, maybe we should switch this out with that and maybe even replace `if wallet.type() ...` with that instead.

File: chia/wallet/trade_manager.py [hunk 3]
```diff
@@ -517,15 +524,26 @@ async def _create_offer_for_ids(
...
                         raise ValueError(f"Cannot offer coins from wallet id {wallet.id()}")
                     # For the XCH wallet also include the fee amount to the coins we use to pay this offer
                     amount_to_select = abs(amount)
                     if wallet.type() == WalletType.STANDARD_WALLET:
                         amount_to_select += fee
-                    coins_to_offer[id] = await wallet.get_coins_to_offer(
-                        asset_id, uint64(amount_to_select), tx_config.coin_selection_config
-                    )
+                    assert isinstance(wallet, (CATWallet, DataLayerWallet, NFTWallet, Wallet))
+                    if isinstance(wallet, DataLayerWallet):
```
Comment (Quexington) [inline chia/wallet/trade_manager.py:535]: I try not to switch on specific wallet types in the trade manager whenever possible.  Probably could have done a better job hinting the `get_coins_to_offer` function in the protocol but I don't see why this change is necessary.

**Review by Quexington — COMMENTED** [review]
The more we could get rid of the `assert isinstance`s referencing anything except `Wallet` would be better I think.

**Review by Quexington — APPROVED** [review]
I don't like adding all of the specific wallet type references here but it's representative of a larger problem that is out of scope.  This is a good step forward.

File: chia/wallet/trade_manager.py [hunk 4]
```diff
@@ -517,15 +524,26 @@ async def _create_offer_for_ids(
...
+                        assert asset_id is not None
+                        coins_to_offer[id] = await wallet.get_coins_to_offer(launcher_id=asset_id)
+                    elif isinstance(wallet, NFTWallet):
+                        assert asset_id is not None
+                        coins_to_offer[id] = await wallet.get_coins_to_offer(nft_id=asset_id)
```
Comment (arvidn) [inline chia/wallet/trade_manager.py:540]: what's the reason for this change? do the wallets not have the same function signature for `get_coins_to_offer()`? It looks like `NFTWallet` and `DataLayerWallet` both take 1 argument, whereas the other take 3.

It seems like that intention was for these to use the same signature, which I would expect to be a more appropriate fix for this

**Review by arvidn — COMMENTED** [review]
shouldn't you remove `trade_manager.py` from the exclusions list too?

File: chia/wallet/trade_manager.py (hunk 4 above)
Comment (AmineKhaldi) [reply chia/wallet/trade_manager.py:540]: Yes, `get_coins_to_offer`'s signature is not the same across wallets, and `asset_id` also differs in terms of optionality. I also used this as a chance to make the param name clearer (`launcher_id` for one wallet and `nft_id` for another).

Comment (arvidn) [reply chia/wallet/trade_manager.py:540]: it seems to go against the original design though; to have these functions implement a common interface.
If we truly want to treat these as different functions, they should probably have different names as well. But I still think the first attempted solution should be to implement the (presumed) original intent. i.e. make them have the same signature. Maybe even enforce the interface with the protocol.

Comment (AmineKhaldi) [reply chia/wallet/trade_manager.py:540]: I share Quex's sentiment that issues like this are representative of a larger problem that is beyond the type annotations scope of this PR.

**Review by arvidn — APPROVED** [review]

**Comment by arvidn** [comment]
I think we can make an exception for the installer test, since it's caused by the missing secrets on a forked repo
